src/mwparserfromhell/parser/ctokenizer/tok_parse.c

```diff
@@ -1558,6 +1558,9 @@ Tokenizer_handle_tag_close_close(Tokenizer *self)
             Py_XDECREF(so);
             Py_XDECREF(sc);
         }
+        // One of `so` or `sc` failed to allocate so we fall through to clean up the
+        // closing tag and propagate the error by returning NULL.
+        __attribute__((fallthrough));
```

Comment:
`__attribute__((fallthrough))` is non-standard, IMO it would be better to use [`[[fallthrough]]`](https://en.cppreference.com/w/c/language/attributes/fallthrough) if C23 is an option, or just duplicate the two lines below (`Py_DECREF` and `return`) to avoid the fallthrough.